runtime/compiler/runtime/RelocationRuntime.cpp

```diff
@@ -665,7 +708,22 @@ TR_RelocationRuntime::relocateMethodMetaData(UDATA codeRelocationAmount, UDATA d
       {
       TR_PersistentJittedBodyInfo *persistentBodyInfo = reinterpret_cast<TR_PersistentJittedBodyInfo *>( (_newPersistentInfo + sizeof(J9JITDataCacheHeader) ) );
       TR_PersistentMethodInfo *persistentMethodInfo = reinterpret_cast<TR_PersistentMethodInfo *>( (_newPersistentInfo + sizeof(J9JITDataCacheHeader) ) + sizeof(TR_PersistentJittedBodyInfo) );
-      persistentBodyInfo->setMethodInfo(persistentMethodInfo);
+
+#if defined(JITSERVER_SUPPORT)
+      if (_comp->isRemoteCompilation() && !_comp->getCurrentMethod()->isInterpreted())
+         {
```

Comment:
I confess that I don't understand what this code is doing, so some comments may be needed here.

Reply:
Found the change introduced in 8ceb77fd3f. Will add more comments from that commit.